spec/miscellaneous/lib/tasks/after_tasks.rake_spec.rb

```diff
@@ -46,3 +46,49 @@
     end
   end
 end
+
+describe "rake After:unhide_invited_works" do
+  let(:invited_anonymous_collection) { create(:anonymous_collection) }
+  let(:invited_unrevealed_collection) { create(:unrevealed_collection) }
+  let(:anonymous_collection) { create(:anonymous_collection) }
+  let(:unrevealed_collection) { create(:unrevealed_collection) }
+  let(:other_collection) { create(:collection) }
+  let(:invited_anonymous_unrevealed_collection) { create(:anonymous_unrevealed_collection) }
+  let(:invited_anonymous_work) { create(:posted_work, collections: [invited_anonymous_collection]) }
+  let(:invited_unrevealed_work) { create(:posted_work, collections: [invited_unrevealed_collection]) }
+  let(:normal_work) { create(:posted_work, collections: [other_collection]) }
+  let(:invited_anonymous_unrevealed_work) { create(:posted_work, collections: [invited_anonymous_unrevealed_collection]) }
+  let(:anonymous_work) { create(:posted_work, collections: [anonymous_collection]) }
+  let(:unrevealed_work) { create(:posted_work, collections: [unrevealed_collection]) }
+
+  context "When invited works are incorrectly anonymous or unrevealed" do
+    before do
+      # Screw up collection items
+      invited_anonymous_work.collection_items.first.update_columns(user_approval_status: 0)
+      invited_unrevealed_work.collection_items.first.update_columns(user_approval_status: 0)
+      invited_anonymous_unrevealed_work.collection_items.first.update_columns(user_approval_status: 0)
+    end
+
+    it "updates the anonymous and unrevealed status of invited work" do
```

Comment:
Should be `works`.

Some whitespace like the other test has might help with readability here -- maybe a couple comments, too:

```
    it "updates the anonymous and unrevealed status of invited work" do
      subject.invoke

      normal_work.reload
      invited_anonymous_work.reload
      invited_unrevealed_work.reload
      invited_anonymous_unrevealed_work.reload
      unrevealed_work.reload
      anonymous_work.reload

      # Accepted works should be unchanged
      expect(normal_work.unrevealed?).to be(false)
      expect(normal_work.anonymous?).to be(false)
      expect(anonymous_work.unrevealed?).to be(false)
      expect(anonymous_work.anonymous?).to be(true)
      expect(unrevealed_work.unrevealed?).to be(true)
      expect(unrevealed_work.anonymous?).to be(false)

      # Invited works should no longer be anonymous or unrevealed
      expect(invited_anonymous_work.unrevealed?).to be(false)
      expect(invited_anonymous_work.anonymous?).to be(false)
      expect(invited_unrevealed_work.unrevealed?).to be(false)
      expect(invited_unrevealed_work.anonymous?).to be(false)
      expect(invited_anonymous_unrevealed_work.anonymous?).to be(false)
      expect(invited_anonymous_unrevealed_work.unrevealed?).to be(false)
    end
```

Probably in an ideal world, these specs would be broken down into something like: 
```
context "when collection is anonymous" do
  it "updates the anonymous status of invited works" do
  end
end

context "when collection is unrevealed" do
  it "updates the unrevealed status of invited works" do
  end
end

context "when collection is unrevealed and anonymous" do
  it "updates the anonymous and unrevealed status of invited works" do
  end
end
```
But I think what you have works well enough for our purposes -- we're not exactly at risk of breaking this task in future code changes and needing to know exactly which scenario is broken.